### PR TITLE
feat(Themes): 2.0 theme

### LINKED ISSUE #15843: Backport Teams 2.0 theme to react-northstar
Backport Teams 2.0 theme to `react-northstar`. (mostly just color scheme/palette changes)

### PR DESCRIPTION
**Pull request by jurokapsiar**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #15843
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

Add 2.0 and 2.0 Dark theme
Added theme picker to color palette and color schemes sites:

Screenshot 2020-12-08 at 10 04 54 AM
Screenshot 2020-12-08 at 10 04 22 AM

### CODE REVIEW

File: packages/fluentui/react-northstar/src/themes/teams-v2/components/Chat/chatMessageVariables.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  actionMenuBoxShadow: siteVars.shadowLevel1,
+  actionMenuPositionRight: pxToRem(5),
+  actionMenuPositionTop: pxToRem(-30),
+  backgroundColor: siteVars.colorScheme.default.background,
+  backgroundColorMine: siteVars.colorScheme.brand.background1,
+  borderRadius: pxToRem(3),
+  color: 'rgb(64, 64, 64)',
+  offset: pxToRem(100),
+  padding: pxToRem(16),
+  authorMarginRight: pxToRem(12),
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/themes/teams-v2/components/Chat/chatMessageVariables.ts:14]: Variable clean up is needed

Comment (ling1726) [reply packages/fluentui/react-northstar/src/themes/teams-v2/components/Chat/chatMessageVariables.ts:14]: Sorry I am stupid/new and didn't understand this comment. Do you mean remove unnecessary variables or clean up wrong values in variables ?

File: packages/fluentui/react-northstar/src/themes/teams-dark-v2/components/Chat/chatMessageVariables.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { ChatMessageVariables } from '../../../teams/components/Chat/chatMessageVariables';
+
+export const chatMessageVariables = (siteVars: any): Partial<ChatMessageVariables> => {
```
Comment (ling1726) [inline packages/fluentui/react-northstar/src/themes/teams-dark-v2/components/Chat/chatMessageVariables.ts:3]: nit: do we not have a type for siteVars ?

File: packages/fluentui/react-northstar/src/themes/teams-v2/components/Chat/chatMessageVariables.ts (hunk 1 above)
Comment (jurokapsiar) [reply packages/fluentui/react-northstar/src/themes/teams-v2/components/Chat/chatMessageVariables.ts:14]: I would also like to understand more about 'variable cleanup' - do you think there are variables that are not used in styles?

Comment (jurokapsiar) [reply packages/fluentui/react-northstar/src/themes/teams-v2/components/Chat/chatMessageVariables.ts:14]: `authorMarginRight` is used in styles. I will check the other ones

Comment (jurokapsiar) [reply packages/fluentui/react-northstar/src/themes/teams-v2/components/Chat/chatMessageVariables.ts:14]: `actionMenuPositionRight` and `actionMenuPositionTop` are not used in styles and can be removed. would you consider removing them a breaking chance @layershifter? they are in the interface

File: packages/fluentui/react-northstar/src/themes/teams-dark-v2/components/Chat/chatMessageVariables.ts (hunk 2 above)
Comment (jurokapsiar) [reply packages/fluentui/react-northstar/src/themes/teams-dark-v2/components/Chat/chatMessageVariables.ts:3]: I haven't found any, also we are importing them just by

```
import * as siteVariables from './siteVariables';
```

Comment (ling1726) [reply packages/fluentui/react-northstar/src/themes/teams-dark-v2/components/Chat/chatMessageVariables.ts:3]: Alright, might make sense to start typing this in the future even just for intellisense/autocomplete.

Doesn't really impact your PR so feel free to resolve

File: packages/fluentui/react-northstar/src/themes/teams-v2/components/Chat/chatMessageVariables.ts (hunk 1 above)
Comment (jurokapsiar) [reply packages/fluentui/react-northstar/src/themes/teams-v2/components/Chat/chatMessageVariables.ts:14]: clarified offline - changed to Partial<...> and removed unnecessary variable overrides

File: packages/fluentui/react-northstar/src/themes/teams-v2/components/Chat/chatVariables.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+export interface ChatVariables {
+  backgroundColor: string;
+}
+
+export const chatVariables = (siteVars): ChatVariables => ({
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/themes/teams-v2/components/Chat/chatVariables.ts:5]:
```suggestion
export const chatVariables = (siteVars): Partial<ChatVariables> => ({
```

File: packages/fluentui/react-northstar/src/themes/teams-v2/components/Chat/chatVariables.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+export interface ChatVariables {
+  backgroundColor: string;
+}
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/themes/teams-v2/components/Chat/chatVariables.ts:3]:
```suggestion
import { ChatVariables } from '../../../teams/components/Chat/chatVariables';
```

**Review by layershifter — APPROVED** [review]

**Review by Ridvanovskyy — APPROVED** [review]

**Comment by codepretty** [comment]
Does it make sense to start adding screener tests on the components we have for the new themes?

We'll want to update other components, but also we'll probably want to eventually decouple the inheritance of v2 themes with current themes.

**Comment by codepretty** [comment]
I don't see the new themes available in the theme dropdown on component pages. Is that something we can add?
image

Can we also have a page for the new palette and schemes like are available for the other themes..
/color-schemes
/color-palette

**Review by codepretty — APPROVED** [review]
Thank you so much for doing this @jurokapsiar 🥇

**Comment by jurokapsiar** [comment]
> Does it make sense to start adding screener tests on the components we have for the new themes?
> 
> We'll want to update other components, but also we'll probably want to eventually decouple the inheritance of v2 themes with current themes.

for tests, yes, it requires some coordination but we should do that

**Review by layershifter — APPROVED** [review]

**Review by codepretty — APPROVED** [review]
Looks great - thank you so much!!

**Pull request merged**